OpenRA.Game/Map/MapPreview.cs

```diff
@@ -140,9 +140,10 @@ public void SetCustomRules(ModData modData, IReadOnlyFileSystem fileSystem, Dict
 							files = files.Append(mapFiles);
 						}
 
+						var stringPool = new HashSet<string>();
```

Comment:
these feel weird, I think we should add in-code comments explaining their existence

Reply:
Added some comments.